Build Error:
```
/mauienv/maui-linux/src/Essentials/src/Accelerometer/AccelerometerQueue.shared.cs(28,4): error CS0131: The left-hand side of an assignment must be a variable, property or indexer [/mauienv/maui-linux/src/Essentials/src/Essentials.csproj::TargetFramework=netstandard2.0]
/mauienv/maui-linux/src/Essentials/src/Accelerometer/AccelerometerQueue.shared.cs(28,4): error CS0131: The left-hand side of an assignment must be a variable, property or indexer [/mauienv/maui-linux/src/Essentials/src/Essentials.csproj::TargetFramework=netstandard2.1]
/mauienv/maui-linux/src/Essentials/src/Accelerometer/AccelerometerQueue.shared.cs(28,4): error CS0131: The left-hand side of an assignment must be a variable, property or indexer [/mauienv/maui-linux/src/Essentials/src/Essentials.csproj::TargetFramework=net8.0-gtk]
/mauienv/maui-linux/src/Essentials/src/Accelerometer/AccelerometerQueue.shared.cs(28,4): error CS0131: The left-hand side of an assignment must be a variable, property or indexer [/mauienv/maui-linux/src/Essentials/src/Essentials.csproj::TargetFramework=net8.0]
    0 Warning(s)
    4 Error(s)
```

--- a/src/Core/src/Platform/Gtk/MauiApplication.cs
+++ b/src/Core/src/Platform/Gtk/MauiApplication.cs
@@ -9,6 +9,12 @@ namespace Microsoft.Maui
 {
 	public abstract class MauiApplication : IPlatformApplication
 	{
+		protected MauiApplication()
+		{
+			Current = this;
+			IPlatformApplication.Current = this;
+		}
+
 		protected abstract MauiApp CreateMauiApp();
 
 		// https://docs.gtk.org/gio/type_func.Application.id_is_valid.html

--- a/src/Core/src/Platform/Gtk/MauiApplication.cs
+++ b/src/Core/src/Platform/Gtk/MauiApplication.cs
@@ -13,7 +13,7 @@ namespace Microsoft.Maui
 
 		// https://docs.gtk.org/gio/type_func.Application.id_is_valid.html
 		// TODO: find a better algo for id
-		public virtual string ApplicationId => $"{typeof(MauiGtkApplication).Namespace}.{nameof(MauiGtkApplication)}.{Name}".PadRight(255, ' ').Substring(0, 255).Trim();
+		public virtual string ApplicationId => $"{typeof(MauiApplication).Namespace}.{nameof(MauiApplication)}.{Name}".PadRight(255, ' ').Substring(0, 255).Trim();
 
 		string? _name;
 
@@ -27,7 +27,7 @@ namespace Microsoft.Maui
 		// https://docs.gtk.org/gtk3/class.Application.html
 		public static Gtk.Application CurrentGtkApplication { get; internal set; } = null!;
 
-		public static MauiGtkApplication Current { get; internal set; } = null!;
+		public static MauiApplication Current { get; internal set; } = null!;
 
 		public MauiGtkMainWindow MainWindow { get; protected set; } = null!;
 

